src/app/events/slash-create/ComponentInteraction.ts

```diff
@@ -0,0 +1,48 @@
+import { DiscordEvent } from '../../types/discord/DiscordEvent';
+import { ComponentContext } from 'slash-create';
+import Log, { LogUtils } from '../../utils/Log';
+import componentInteractionSquadUp from './squad/ComponentInteractionSquadUp';
+
+export interface ComponentMeta {
+	feature: string,
+	method:string,
+	label:string,
+	data?:string
+}
+
+export default class implements DiscordEvent {
+
+	name = 'componentInteraction';
+	once = false;
+
+	formatComponentMetaData(componentContext: ComponentContext): ComponentMeta {
+
+		const componentMeta = {} as ComponentMeta;
+		try {
+			[componentMeta.feature, componentMeta.method, componentMeta.label, componentMeta.data] = componentContext.customID.split(':');
+		} catch (e) {
+			// do nothing 
```

Comment:
Can we log an `info` or `debug` statement in the event of an error? Or pass the error to another handler?

Reply:
So this one is something we should maybe discuss to come up with a standard for future development. In this example it would not be supposed to error unless you don't stick to that 'standard' in which case it would not really be an error. That is why I didn't log it. But for context: The only way to capture "non-discord" or "application specific" information during the button click event is by passing that info as the `customId<string>` of the button (which you get in the context).  Above I just used a `:` delimited string with the format `featureName:functionName:buttonLabel:metaData`. But it might just as well be possible to format it in json and do JSON.parse instead. If we would agree on a standard format to use throughout the app we can have it parsed in this step and then always have the same object format. That is just a suggestion though and we can also leave the parsing to individual feature implementations instead.

Reply:
Gotcha. It makes that we establish a standard for logging events. We have an inordinate amount of errors being logged, and the bot reboots regularly, so having more information about what's happening where from a quick glance through logs will be super helpful. I'll start a thread about it, as the sooner we come to terms on this, the sooner we can make the upgrades and implementations.

Go ahead and leave the logging as-is, as you've coded it, and we'll update accordingly when we come to a decision on how we want to approach this.